tools/pre_ci.py

```
@@ -0,0 +1,201 @@
+        False: "❌ FAIL",
+        "SKIPPED": "⏭️  SKIP",
+    }
+
+    def __init__(self, min_ver: str = "3.10", max_ver: str = "3.14"):
+        """Initializes the PreCIPipeline with Python version bounds."""
+        self._results = []
+        self.is_ci = os.environ.get("CI", "").lower() in ("true", "1", "yes")
+        self.min_ver = min_ver
+        self.max_ver = max_ver
+
+    def record_result(self, description: str, passed: bool | str) -> None:
```

Ruff (0.15.11)

[warning] Boolean-typed positional argument in function definition

(FBT001)